From f5d1d7e683cbb9b8cba1ecb01a3caf213afc017b Mon Sep 17 00:00:00 2001
From: Wim Taymans <wtaymans@redhat.com>
Date: Wed, 9 Aug 2023 12:45:18 +0200
Subject: [PATCH 5/5] evl: fix timerfd read

It should read uint64_t now.

Tested: Jorge Ramirez-Ortiz <jorge@foundries.io>
---
 spa/plugins/support/evl-system.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/spa/plugins/support/evl-system.c b/spa/plugins/support/evl-system.c
index 1f4381a67..336b1080b 100644
--- a/spa/plugins/support/evl-system.c
+++ b/spa/plugins/support/evl-system.c
@@ -249,10 +249,8 @@ static int impl_timerfd_gettime(void *object,
 }
 static int impl_timerfd_read(void *object, int fd, uint64_t *expirations)
 {
-	uint32_t ticks;
-	if (oob_read(fd, &ticks, sizeof(ticks)) != sizeof(ticks))
+	if (oob_read(fd, expirations, sizeof(uint64_t)) != sizeof(uint64_t))
 		return -errno;
-	*expirations = ticks;
 	return 0;
 }
 
-- 
2.34.1

